### ISSUE TITLE
rethink imagestatus?

### ISSUE THREAD

**Issue by cpuguy83** [eraser-dev/eraser#96]
Today we keep track of the result of an `imagejob` by creating an `imagestatus` for each node.
The simple version of this looks like:

```mermaid
graph TD;
ImageList1 --> ImageJob1;
ImageJob1 --> EraserPodJob1-Node1;
ImageJob1 --> EraserPodJob1-Node2;
ImageJob1 --> EraserPodJob1-NodeN;
EraserPodJob1-Node1 --> ImageStatus-Node1;
EraserPodJob1-Node2 --> ImageStatus-Node2;
EraserPodJob1-NodeN --> ImageStatus-NodeN;
ImageStatus-Node1 --> ImageJob1;
ImageStatus-Node2 --> ImageJob1;
ImageStatus-NodeN --> ImageJob1;
ImageJob1 --> ImageList1;
```

However if you add a 2nd `imagelist` to the mix, you may notice that `imagestatus` is per node, not per job per node.
This is significantly more complex and likely to cause issues:

```mermaid
graph TD;
ImageList1 --> ImageJob1;
ImageJob1 --> EraserPodJob1-Node1;
ImageJob1 --> EraserPodJob1-Node2;
ImageJob1 --> EraserPodJob1-NodeN;
EraserPodJob1-Node1 --> ImageStatus-Node1;
EraserPodJob1-Node2 --> ImageStatus-Node2;
EraserPodJob1-NodeN --> ImageStatus-NodeN;
ImageStatus-Node1 --> ImageJob1;
ImageStatus-Node2 --> ImageJob1;
ImageStatus-NodeN --> ImageJob1;
ImageJob1 --> ImageList1;
ImageList2 --> ImageJob2;
ImageJob2 --> EraserPodJob2-Node1;
ImageJob2 --> EraserPodJob2-Node2;
ImageJob2 --> EraserPodJob2-NodeN;
EraserPodJob2-Node1 --> ImageStatus-Node1;
EraserPodJob2-Node2 --> ImageStatus-Node2;
EraserPodJob2-NodeN --> ImageStatus-NodeN;
ImageStatus-Node1 --> ImageJob2;
ImageStatus-Node2 --> ImageJob2;
ImageStatus-NodeN --> ImageJob2;
ImageJob2 --> ImageList2;
```

Even if we do make this per job per node, this becomes significantly cleaner and less error prone, with each imagelist effectively creating a separate tree of events as one might expect:

```mermaid
graph TD;
ImageList1 --> ImageJob1;
ImageJob1 --> EraserPodJob1-Node1;
ImageJob1 --> EraserPodJob1-Node2;
ImageJob1 --> EraserPodJob1-NodeN;
EraserPodJob1-Node1 --> ImageStatusJob1-Node1;
EraserPodJob1-Node2 --> ImageStatusJob1-Node2;
EraserPodJob1-NodeN --> ImageStatusJob1-NodeN;
ImageStatusJob1-Node1 --> ImageJob1;
ImageStatusJob1-Node2 --> ImageJob1;
ImageStatusJob1-NodeN --> ImageJob1;
ImageJob1 --> ImageList1;
ImageList2 --> ImageJob2;
ImageJob2 --> EraserPodJob2-Node1;
ImageJob2 --> EraserPodJob2-Node2;
ImageJob2 --> EraserPodJob2-NodeN;
EraserPodJob2-Node1 --> ImageStatusJob2-Node1;
EraserPodJob2-Node2 --> ImageStatusJob2-Node2;
EraserPodJob2-NodeN --> ImageStatusJob2-NodeN;
ImageStatusJob2-Node1 --> ImageJob2;
ImageStatusJob2-Node2 --> ImageJob2;
ImageStatusJob2-NodeN --> ImageJob2;
ImageJob2 --> ImageList2;
```

However then we have this extra object that is effectively reporting exactly what the pod already encapsulates since we'd have 1 `imagestatus` per `imagelist` per node.
Which we only end up collecting to throw into `imagelist`'s status anyway.

At this point I also have to wonder if anyone actually needs the detailed status.
On success of a job then the inventory of images on a node will be updated by virtue of that success.
On failure a status is useful for debugging purposes, however we can keep the `imagejob` (and by extension the failed pods for the job) so an admin can look at pod logs and whatever else we end up providing.

To help surface details, the job controller can emit events (for humans) and/or metrics on success or failure.
Such metrics would really only be to help surface errors on the cluster since an image removal error is likely indicative of other problems.
Given that there is one `imagejob` per `imagelist` at a time, the `imagejob` can post the last status in the `imagelist` as it does today (where status would be at most a summary + last run timestamp).

Currently this also means the eraser pod requires cluster access so it can post an `imagestatus` (it also does to get the `imagelist` spec that it is running for... but this can be solved in other ways). Ideally this pod does not require any special privileges except on the node itself to communicate with CRI.

----

TLDR; I think we can get rid of `imagestatus` and the current implementation of `imagestatus` is error prone as it is. Removing `imagestatus` is also a step towards not requiring any cluster privileges for the eraser pod.
